### DISCUSSION TITLE
Suggestion: don't show `[default: None]` in help for required options

### DISCUSSION THREAD

**Discussion by aldanor** [fastapi/typer#693] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def main(x: Annotated[int, typer.Option()]):
    pass
```

```sh
> python foo.py --help
...
│ *  --x           INTEGER  [default: None] [required]                                                                                         
...
```

### Description

If provided with an option without a default, it:
- makes it required and displays as `[required]` in help (good)
- still shows `[default: None]` in help (not good) which is very confusing since this default will never be used and conflicts with the 'required' label for this option

Suggestion: remove `[required]` label in help for required options. (yes, there's `show_default` but does it make sense having it default to True in this case?)

### Operating System

macOS

### Typer Version

0.9.0

### Python Version

3.11.5

**Comment by YuriiMotov** [comment]
Seems to be fixed:
```
typer myapp.py run --help
                                                                                                                                                         
 Usage: typer [PATH_OR_MODULE] run [OPTIONS]                                                                                                             
                                                                                                                                                         
 Run the provided Typer app.                                                                                                                             
                                                                                                                                                         
╭─ Options ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ *  --x           INTEGER  [required]                                                                                                                  │
│    --help                 Show this message and exit.                                                                                                 │
╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```
